opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Dict, Union
+from opentelemetry.resources import Resource
+
+
+class DefaultResource(Resource):
```

Comment:
I suggest that we name this class as `Resource` similar like what @c24t did for [`Span`](https://github.com/open-telemetry/opentelemetry-python/blob/master/opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py#L144).

This makes sure a much cleaner callstack (I would struggle with name `DefaultResource` and `DefaultSpan` #63).

Reply:
Got it, I can perform that pattern.

Maybe a separate issue/pr since there's already a precedent, but I think it would still be beneficial to differentiate between the API and the Default. Maybe ResourceABC and Resource? or ResourceInterface?

Reply:
That's a valid point, and it could be helpful if some error message only has class name but not the full module path.
In Java and .NET world I've seen something like Resource and ResourceImpl (I'm trying to avoid IResource here since Python doesn't really have interface concept). We can probably steal the idea (or peek at what other Python libs are doing).

Reply:
if we go really old school we can follow the zope interface convention (and add the I):

https://zopeinterface.readthedocs.io/en/latest/README.html#defining-interfaces

I'll fill a separate issue for that, so we can carry that discussion separately.